Android: Set pending intent flag to stop insta-crash

phil-hudson opened the pull request:
To fix https://github.com/godotengine/godot/issues/78105

Specifically, setting the flag for PendingIntent to fix error:

`Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.`

phil-hudson commented:
> For the record, your commit seems not to be linked to your GitHub account. See: [Why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) for more info.
> 
> Specifically here, you didn't fill in an email to associate with your Git commit, so the author info is invalid (see header of https://patch-diff.githubusercontent.com/raw/godotengine/godot/pull/78175.patch ).

thanks for that, should be resolved now.

akien-mga commented:
You would need to amend the commit and force push to your remote to fix it properly.
```
git commit --amend --reset-author
git push --force
```

phil-hudson committed:
fix(android): set pending intent flag to stop insta-crash

fix(android): add conditional check for minimum sdk version

fix(android): formatting

fix(android): formatting

fix(android): formatting

fix(android): formatting again

m4gr3d commented on platform/android/java/lib/src/org/godotengine/godot/Godot.java:
```diff
@@ -712,7 +712,7 @@ public void onCreate(Bundle icicle) {
 
 				PendingIntent pendingIntent = PendingIntent.getActivity(activity, 0,
-						notifierIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+						notifierIntent, PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
```
The added flag was introduced in api 23 and our min is 21, so you need to add a conditional check prior to using it 

https://developer.android.com/reference/android/app/PendingIntent#FLAG_IMMUTABLE

m4gr3d reviewed (approved):
Looks good!

Can you squash the commits into a single one.

phil-hudson commented:
squashed and pushed!

YuriSizov commented:
Cherry-picked for 4.0.4.
